config/pmd.xml

```diff
@@ -17,10 +17,51 @@
     <exclude name="StdCyclomaticComplexity"/>
   </rule>
 
+  <rule ref="rulesets/java/codesize.xml/NcssCount">
+    <properties>
+      <property name="methodReportLevel" value="32"/>
+      <!-- The classes Main, PackageObjectFactory, CheckstyleAntTask, RequireThisCheck,
+             VariableDeclarationUsageDistanceCheck, CommentsIndentationCheck, JavadocMethodCheck,
+             HandlerFactory are big and should not be split without a good reason.
+           The classes CheckerTest, ConfigurationLoaderTest, MainTest, TreeWalkerTest,
+             CheckstyleAntTaskTest, CustomImportOrderCheckTest, ImportOrderCheckTest,
+             IndentationCheckTest, XdocsPagesTest, XpathMapperTest, GeneratedJavadocTokenTypesTest
+             are test classes and should be one whole piece of code.
+           The methods visitToken/leaveToken are a big SWITCH block with a nuber of IF blocks.
+             If we split the block to several methods it will damage the readability.
+           The methods XMLLogger.encode, FallThroughCheck.isTerminated, ElementNode.iterateAxis,
+             NoWhitespaceAfterCheck.getArrayDeclaratorPreviousElement are also huge switches,
+             they has to be monolithic.
+           The methods JavadocTokenTypesTest.testTokenValues and
+             GeneratedJavaTokenTypesTest.testTokenNumbering contains several asserts as they
+             checks each token explicitly. -->
+      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main'
+        or @Image='PackageObjectFactory' or @Image='CheckstyleAntTask' or @Image='RequireThisCheck'
+        or @Image='VariableDeclarationUsageDistanceCheck' or @Image='CommentsIndentationCheck'
+        or @Image='JavadocMethodCheck' or @Image='CheckerTest' or @Image='ConfigurationLoaderTest'
+        or @Image='MainTest' or @Image='TreeWalkerTest' or @Image='CheckstyleAntTaskTest'
```

Comment:
I assume there is no way to match all test classes?

Reply:
Done